File: backend/app/schemas/financials.py
from typing import Optional

from pydantic import BaseModel


class VendorOut(BaseModel):
    id: int
    name: str
    billing_type: Optional[str] = None
    hourly_rate: Optional[float] = None
    role_key: Optional[str] = None
    active: Optional[int] = 1


class InvoiceOut(BaseModel):
    id: int
    month: str  # "YYYY-MM"
    msa_amount: float = 0.0
    tm_amount: float = 0.0
    total_amount: float = 0.0
    invoice_number: Optional[str] = None
    received_date: Optional[str] = None
    paid: int = 0
    notes: Optional[str] = None


class MonthlyCostOut(BaseModel):
    month: str
    billing_type: str
    total_hours: float = 0.0
    total_cost: float = 0.0


class ProjectCostOut(BaseModel):
    ete_project_id: Optional[str] = None
    ete_project_name: Optional[str] = None
    project_hours: float = 0.0
    support_hours: float = 0.0
    total_hours: float = 0.0
    total_cost: float = 0.0
    tm_cost: float = 0.0
    msa_hours: float = 0.0
    tm_hours: float = 0.0

File: frontend/src/types/financials.ts
export interface Vendor {
  id: number;
  name: string;
  billing_type: string | null;
  hourly_rate: number | null;
  role_key: string | null;
  active: number;
}

export interface Invoice {
  id: number;
  month: string;
  msa_amount: number;
  tm_amount: number;
  total_amount: number;
  invoice_number: string | null;
  received_date: string | null;
  paid: number;
  notes: string | null;
}

export interface MonthlyCost {
  month: string;
  billing_type: string;
  total_hours: number;
  total_cost: number;
}

export interface ProjectCost {
  ete_project_id: string | null;
  ete_project_name: string | null;
  project_hours: number;
  support_hours: number;
  total_hours: number;
  total_cost: number;
  tm_cost: number;
  msa_hours: number;
  tm_hours: number;
}
